### DISCUSSION TITLE
Use of `pubh::contingency()` affects appearance on subsequent slides

### DISCUSSION THREAD

**Discussion by snhansen** [quarto-dev/quarto-cli#5996] [Q&A] [answered]
### Description

Consider the example below. The appearance of the dataset printed on the second slide is affected by the code run on the first. Specifically, if you remove the code on the first slide the data on the second is shown correctly. What's going on here?

	---
	format: revealjs
	---

	## 1

	```{r}
	library(pubh)
	mtcars |>
	  contingency(as.factor(vs) ~ as.factor(am))
	```

	## 2

	```{r}
	mtcars
	```

Since `pubh::contingency` is a wrapper for `epiR::epi.2by2` I tried replacing the first slide with

	## 1

	```{r}
	library(epiR)
	epi.2by2(c(7,6,7,12))
	```

which works. So somehow the `contingency`-function is playing games on me but it shouldn't really be able to affect subsequent slides, should it? I'm on Quarto v. 1.4.130.

**Comment by mcanouil** [comment]
It seems to be only print/cat calls.
It's not just a wrapper to the function you mentioned, there is also `epitools::editable`.

What's the output exactly?

**Comment by cderv** [comment, marked as answer]
First, This happens also is you use `html` format. This means that is not a `revealjs` problem.

Then, it happens if you only load the `pubh` package. 
````markdown
---
format: html
---

## 1

```{r}
#| message: false
library(pubh)
```

## 2

```{r}
mtcars
```
````

This gives us the hint that the package **pubh** is loading something that changes the default behavior. 

Inspecting the result table and what the package is loading, it seems that the output are **huxtable** output. We can reproduce now with only this package

````markdown
---
format: html
---

## 1

```{r}
#| message: false
library(huxtable)
```

## 2

```{r}
mtcars
```
````

And it happens this function is loading a `huxtable:::knit_print.data.frame()` method. This means that every data.frame output will be process through this function. It is documented here: https://hughjonesd.github.io/huxtable/reference/knit_print.data.frame.html
They explain you can deactivate using an option `options(huxtable.knit_print_df = FALSE)`

Settings this works to deactivate the behavior. 

You can maybe report to **pubh** (https://github.com/josie-athens/pubh) if they don't document this. 

Hope it helps understand the behavior. (Not Quarto related, it would be the same in R Markdown). I tried to share my search process so that you can know how to find this kind of understanding next time ! 😉

**Reply by snhansen** [reply]
That's some fine detective work, @cderv. Thank you for the detailed response.
